### PR TITLE
Add clippy CI check to workflow

### LINKED ISSUE #75: Add `clippy` CI check

### PR DESCRIPTION
**Pull request by andrewvious**
Closes #75 

The clippy check isn't currently passing, working on getting the warnings taken care of in a separate PR.

### CODE REVIEW

**Comment by andrewvious** [comment]
There are only two clippy warnings left revolving around a `manual implementation of an assigned operation`.
These warning's can be handled in a separate PR need be @eureka-cpu @mitch-vrrb

**Comment by mitch-vrrb** [comment]
Those clippy errors aren't warnings, this test overflows the stack:


```
#[cfg(test)]
mod test {
    #[test]
    fn test_u256_into() {
        let eth_u256: ethereum_types::U256 = ethereum_types::U256::one();

        let token_u256: &mut super::U256 = eth_u256.into();
        assert_eq!(eth_u256, token_u256.into());
    }
}
```

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -4,6 +4,31 @@ env:
...
+  clippy:
+    name: clippy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dev-dependencies
+        run: sudo apt-get install -y libxcb-shape0-dev libxcb-xfixes0-dev
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          submodules: true
```
Comment (eureka-cpu) [inline .github/workflows/rust.yml:17]: I think you can remove this

**Review by eureka-cpu — CHANGES_REQUESTED** [review]
Can you see if removing the recursive `From` implementations causes any failures? Left one other review comment, otherwise looks good.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -4,6 +4,24 @@ env:
...
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: cargo clippy
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-features --all-targets -- -D warnings
```
Comment (eureka-cpu) [inline .github/workflows/rust.yml:23]: Seems it's having trouble finding the `Cargo.toml` file. Try adding `--workspace` to this and see if it fixes the failure.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (andrewvious) [reply .github/workflows/rust.yml:17]: I added this chunk back, and clippy is able to find the `Cargo.toml` again.

Comment (eureka-cpu) [reply .github/workflows/rust.yml:17]: Try adding this instead, maybe it has to do with the fact we aren't installing a rust toolchain which might be required by the actions-rs clippy-check 
```
- name: Install toolchain
  uses: dsherret/rust-toolchain-file@v1
```

**Pull request merged**